File: 2023/day1/src/part1.ts
import fs from 'fs';

function main() {
  const file = fs.readFileSync('./bin/input.txt').toString();

  const lines = file.split('\n');
  let sum = 0;

  for (const line of lines) {
    let i = 0,
      j = line.length - 1;

    while (i <= j) {
      if (!isNaN(Number(line[i])) && !isNaN(Number(line[j]))) {
        break;
      } else {
        if (isNaN(Number(line[i]))) {
          ++i;
        }
        if (isNaN(Number(line[j]))) {
          --j;
        }
      }
    }

    sum += parseInt(`${line[i]}${line[j]}`);
  }

  console.log(sum);

  return sum;
}

main();

File: 2023/day1/src/bin/part1.rs
fn main() {
    let content = include_str!("input.txt");

    let mut sum: u32 = 0;

    for line in content.lines() {
        let mut i = 0;
        let mut j = line.len() - 1;

        let chars: Vec<char> = line.chars().collect();

        while i <= j {
            if chars[i].is_numeric() && chars[j].is_numeric() {
                break;
            } else {
                if chars[i].is_alphabetic() {
                    i += 1
                }
                if chars[j].is_alphabetic() {
                    j -= 1
                }
            }
        }

        sum += format!("{}{}", chars[i], chars[j]).parse::<u32>().unwrap();
    }

    println!("{sum}");
}
